modules/packages/LinearAlgebra.chpl

```diff
@@ -340,64 +342,72 @@ class ExpmPadeHelper {
     return this.exactD10;
   }
 
-  proc comp_looseD4(){
-    if this.useExactOneNorm then{
+  proc comp_looseD4() {
+    if this.useExactOneNorm || isComplexType(this.eltType) then {
       return comp_exactD4();
     }
     else {
       if !isApproxdComputed["A4"] then {
-        //TODO: Need to write a function to compute the
-        // estimated norm (change the below function)
-        // Right now, just returns the exact norm
-        this.approxD4 = norm(comp_A4(), normType.norm1)**(1/4.0);
+        try {
+          this.approxD4 = onenormest(comp_A4())**(1/4.0);
+        }
+        catch e{
+          this.approxD4 = comp_exactD4();
+        }
         isApproxdComputed["A4"] = true;
       }
       return this.approxD4;
     }
   }
 
-  proc comp_looseD6(){
-    if this.useExactOneNorm then{
+  proc comp_looseD6() {
+    if this.useExactOneNorm || isComplexType(this.eltType) then {
       return comp_exactD6();
     }
     else {
       if !isApproxdComputed["A6"] then {
-        //TODO: Need to write a function to compute the
-        // estimated norm (change the below function)
-        // Right now, just returns the exact norm
-        this.approxD6 = norm(comp_A6(), normType.norm1)**(1/6.0);
+        try {
+          this.approxD4 = onenormest(comp_A6())**(1/6.0);
```

Comment:
Here and below, I think these are meant to be `approxD6`, `approxD8` etc. 

This is a good example where the refactorization that @e-kayrakli suggested would be useful.